msal/application.py

```diff
@@ -623,6 +624,9 @@ def __init__(
         # Here the self.authority will not be the same type as authority in input
         if oidc_authority and authority:
             raise ValueError("You can not provide both authority and oidc_authority")
+        if isinstance(authority, str) and urlparse(authority).path.startswith(
+            "/dstsv2"):  # dSTS authority's path always starts with "/dstsv2"
+            oidc_authority = authority  # So we treat it as if an oidc_authority
```

Comment:
+1

I am wondering though how does MSAL Py handle changing the tenant at the request level? I am asking because Azure SDK does this and Azure SDK wants to remain agnostic of authority type (i.e. they want MSAL to just work with AAD, ADFS, dSTS)

Reply:
The simplest is to have `request.WithTenantId(tid)` do nothing in case the authority is dSTS. 

Needs a unit test.

Reply:
[MSAL Python does not currently support `request.WithTenant(...)`](https://msal-python.readthedocs.io/en/latest/#msal.ConfidentialClientApplication.acquire_token_for_client) at all. Confirmed with a downstream partner that they are not currently using it either. It was tracked as a feature request, which we can work on.

Do we want to proceed with this PR as-is, or mark it as blocked while we pick up the `WithTenant(...)` feature now?